searx/engines/ask.py

```diff
@@ -0,0 +1,78 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""Ask.com"""
+
+from urllib.parse import urlencode
+import dateutil
+from lxml import html
+from searx import utils
+
+# Metadata
+about = {
+    "website": "https://www.ask.com/",
+    "wikidata_id": 'Q847564',
+    "official_api_documentation": None,
+    "use_official_api": False,
+    "require_api_key": False,
+    "results": "HTML",
+}
+
+# Engine Configuration
+categories = ['general']
+paging = True
+
+# Base URL
+base_url = "https://www.ask.com/web"
+
+
+def request(query, params):
+
+    query_params = {
+        "q": query,
+        "page": params["pageno"],
+    }
+
+    params["url"] = f"{base_url}?{urlencode(query_params)}"
+    return params
+
+
+def response(resp):
+
+    start_tag = 'window.MESON.initialState = {'
+    end_tag = '}};'
+
+    dom = html.fromstring(resp.text)
+    script = utils.eval_xpath_getindex(dom, '//script', 0, default=None).text
+
+    pos = script.index(start_tag) + len(start_tag) - 1
+    script = script[pos:]
+    pos = script.index(end_tag) + len(end_tag) - 1
+    script = script[:pos]
```

Comment:
```suggestion
    start_pos = script.index(start_tag) + len(start_tag) - 1
    end_pos = script.index(end_tag) + len(end_tag) - 1
    script = script[start_pos:end_pos]
```

Reply:
> call has to read less characters?

My version asserts that the `end_tag` is not taken from a string before `start_tag` / assume start_tag is `= {`  end_tag is `};` .. then in your version the first `end_tag` is in the first line but the `start_tag` is in the second line 

```
 window.MESON = window.MESON || {};
 window.MESON.initialState = {"siteConfig": ...
  ...}};
```

Reply:
Oh, that makes sense :+1: